### ISSUE TITLE
Docker-compose container doesn't connect to IRC over IPv6 and times out

### ISSUE THREAD

**Issue by zejar** [42wim/matterbridge#1129]
**Describe the bug**
So I just stumbled onto this very strange issue. When I run the container using `docker run` the container gets an IPv6 address and can connect to IRC successfully, but when started using docker-compose it times out.

**To Reproduce**
The working docker run command is: `docker run -v /home/user/docker/matterbridge/matterbridge.toml:/matterbridge.toml 42wim/matterbridge -debug`
The compose file that times out:
```
version: '2.4'

services:        
  matterbridge:
    image: 42wim/matterbridge
    restart: unless-stopped
    container_name: matterbridge
    command: "-debug"
    volumes:
      - ${DOCKER_DIR}/matterbridge/matterbridge.toml:/matterbridge.toml
    networks:
      default:
        ipv6_address: 2001:db8:123:d008:4:455c:75c9:2818

networks:
  default:
    enable_ipv6: true
    driver: bridge
    ipam:
      driver: default
      config:
      - subnet: 2001:db8:123:d008:4::/80
        gateway: 2001:db8:123:d008:4::1
```

After that `docker-compose up -d`.

**Expected behavior**
It is expected to connect to the IRC server as it would when run using `docker run`. It can also successfully connect to other sites using IPv6, I've ran this setup for a few years now with other containers.

**Screenshots/debug logs**
```
[0000]  INFO main:         Enabling debug logging.
[0000]  INFO main:         Running version 1.17.5-dev 52a071e3
[0000]  INFO main:         WARNING: THIS IS A DEVELOPMENT VERSION. Things may break.
[0000]  INFO router:       Parsing gateway gateway1
[0000]  INFO router:       Starting bridge: irc.myirc
[0000]  INFO irc:          Connecting irc.snt.utwente.nl:6697
[0005] ERROR irc:          disconnect: error: dial tcp: i/o timeout
[0005] FATAL main:         Starting gateway failed: Bridge irc.myirc failed to start: connection failed dial tcp: i/o timeout
```

This timeout message appears fairly quick after starting the container. When running a tcpdump on the host I don't see any traffic going out to the IRC server, but I do see outgoing traffic to Telegram (the other protocol configured) and I also see dns requests for the Telegram API.

**Environment (please complete the following information):**
 - OS: Debian 10.3
 - Matterbridge version: 1.17.5-dev
 - If self compiled: output of `git rev-parse HEAD`

**Additional context**
```
[irc.myirc]
Nick="<nick>"
NickServNick="<nickname>"
NickServPassword="<password>"
Server="irc.snt.utwente.nl:6697"
UseTLS=true
UseSASL=true
SkipTLSVerify=false
RemoteNickFormat=""

[telegram.mytelegram]
Token="<token>"
RemoteNickFormat="<{NICK}> "

[[gateway]]
name="gateway1"
enable=true

[[gateway.inout]]
account="irc.myirc"
channel="#<channel>"

[[gateway.inout]]
account="telegram.mytelegram"
channel="<channel>"
```

**Comment by qaisjp** [comment]
Have you seen https://docs.docker.com/config/daemon/ipv6/ ?

**Comment by zejar** [comment]
Hi @qaisjp, yes I have. Outgoing network traffic (both IPv6 and IPv4) works fine, I can see the container requesting dns records (over IPv6) for the Telegram API and it also connects to the Telegram API (also over IPv6).
Other containers on the same bridge work perfectly fine with IPv6 too. It just doesn't request dns records for irc.snt.utwente.nl and it never makes outgoing requests to the irc server.
Even if IPv6 wasn't working it would be able to fall back to IPv4 (the docker-compose does work with an IPv4 bridge for some reason).
The thing that I find really strange is that when I use the docker run command it also automatically gets an IPv6 address (the prefix is configured in the /etc/docker/daemon.json file) but then it can connect to the IRC server (I haven't ran across this problem with all of the other containers I'm running).

**Comment by 42wim** [comment]
Timeout seems like a firewall issue.
Sorry but debugging docker issues is out of scope for the project. 
You can always just run the binary, no need for a docker container.

**Comment by zejar** [comment]
In case anyone stumbles across this problem in the future, I did manage to find a workaround for this issue.
Turns out it wasn't a firewall issue, but the usage of a named Docker network.
When the default `bridge` gets used (also configured with IPv6) for the matterbridge container it is able to connect with IRC servers over IPv6. Only when a named Docker network gets used matterbridge will fail to connect to IRC servers over IPv6.
